cli/src/commands/extensions/mod.rs

```diff
@@ -273,7 +274,24 @@ fn ask_permissions(extension: &Extension) -> Result<()> {
         }
     }
 
-    print_permissions_list("Read", "path", permissions.read.get());
+    // Calculate effective read permissions
+    let read = match (&permissions.read, &permissions.run) {
+        (Permission::Boolean(true), _) => Permission::Boolean(true),
+        (_, Permission::Boolean(true)) => Permission::Boolean(true),
```

Comment:
```suggestion
        (Permission::Boolean(true), _) | (_, Permission::Boolean(true)) => Permission::Boolean(true),
```

I don't really mind it either way, but this would technically reduce the duplicate `Permission::Boolean(true)` and thus make it easier to change or add new branches in the future.

Reply:
`rustfmt` split the line because it's too long after making this change. But it's still probably better this way.

I suppose we could do the same thing for the `false` case. That is:

```
        (Permission::Boolean(false), other) | (other, Permission::Boolean(false)) => other.clone(),
```

But binding on both sides of an `|` might be confusing to read.